libs/application/templates/driving-license/src/dataProviders/EligibilityProvider.ts

```diff
@@ -0,0 +1,75 @@
+import { ApplicationEligibilityRequirement } from '@island.is/api/schema'
+import {
+  BasicDataProvider,
+  Application,
+  SuccessfulDataProviderResult,
+  FailedDataProviderResult,
+} from '@island.is/application/core'
+
+const extractReason = (
+  requirements: ApplicationEligibilityRequirement[],
+): string => {
+  return requirements
+    .filter(({ requirementMet }) => !requirementMet)
+    .map(({ key }) => key)
```

Comment:
If these reasons are displayed to the user we should consider mapping them to values in messages to support multiple languages